File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/models/search/searchbook/SearchBookEntityMapper.java
package com.hgianastasio.biblioulbrav2.data.models.search.searchbook;

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper;
import com.hgianastasio.biblioulbrav2.core.search.models.SearchBook;

import java.text.ParseException;
import java.text.SimpleDateFormat;

import javax.inject.Inject;

/**
 * Created by heitor_12 on 09/05/17.
 */

public class SearchBookEntityMapper extends Mapper<SearchBook, SearchBookEntity> {
    @Inject public SearchBookEntityMapper() {}

    @Override
    public SearchBookEntity transform(SearchBook input) {
        throw new IllegalStateException("Not implemented");
    }

    @Override
    public SearchBook transformBack(SearchBookEntity input) {
        SearchBook output = new SearchBook();
        output.setAuthor(input.getAuthor());
        output.setCatchedExps(input.getCatchedExps().split("(?<=\\))"));
        output.setNumber(Long.parseLong(input.getNumber()));
        output.setTitle(input.getTitle());
        try {
            output.setYear(new SimpleDateFormat("yyyy").parse(input.getYear()));
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return output;
    }

}

File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/models/search/searchbook/SearchBookEntityMapper.kt
package com.hgianastasio.biblioulbrav2.data.models.search.searchbook

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper
import com.hgianastasio.biblioulbrav2.core.search.models.SearchBook
import java.text.ParseException
import java.text.SimpleDateFormat
/**
 * Created by heitor_12 on 09/05/17.
 */
class SearchBookEntityMapper constructor() : Mapper<SearchBook, SearchBookEntity>() {
    override fun transform(input: SearchBook): SearchBookEntity? {
        throw IllegalStateException("Not implemented")
    }

    override fun transformBack(input: SearchBookEntity): SearchBook? {
        val output = SearchBook()
        output.author = input!!.author
        output.catchedExps = input.catchedExps?.split("(?<=\\))".toRegex())?.toTypedArray()
        output.number = input.number!!.toLong()
        output.title = input.title
        try {
            output.year = SimpleDateFormat("yyyy").parse(input.year)
        } catch (e: ParseException) {
            e.printStackTrace()
        }
        return output
    }
}